### PR TITLE
bpf: Add macros for sk_skb programs

### PR DESCRIPTION
**Pull request by dave-tucker**
This commit adds the stream_parser and stream_verdict macros for use in
bpf programs

Signed-off-by: Dave Tucker <dave@dtucker.co.uk>

### CODE REVIEW

File: bpf/aya-bpf-macros/src/expand.rs [hunk 1]
```diff
@@ -441,3 +441,53 @@ impl BtfTracePoint {
...
+        } else {
+            format!("sk_skb/{}", kind)
+        };
+        let fn_name = &self.item.sig.ident;
+        let item = &self.item;
+        Ok(quote! {
+            #[no_mangle]
+            #[link_section = #section_name]
+            fn #fn_name(ctx: *mut ::aya_bpf::bindings::__sk_buff) -> u32 {
+                return #fn_name(::aya_bpf::programs::SkSkbContext::new(ctx));
```
Comment (willfindlay) [inline bpf/aya-bpf-macros/src/expand.rs:487]: I know StreamVerdict programs are expected to return non-zero values but is this true for StreamParsers as well? If not, maybe we should match on `self.kind` here and ignore the return value for StreamParsers.

Comment (dave-tucker) [reply bpf/aya-bpf-macros/src/expand.rs:487]: I believe that StreamVerdict programs are supposed to return the number of bytes read.
Here's an (old) example from the kernel source https://elixir.bootlin.com/linux/v4.14.46/source/samples/sockmap/sockmap_kern.c#L49

Comment (willfindlay) [reply bpf/aya-bpf-macros/src/expand.rs:487]: Cool, makes sense.

Comment (dave-tucker) [reply bpf/aya-bpf-macros/src/expand.rs:487]: yes i do lol

**Review by alessandrod — APPROVED** [review]

**Review by willfindlay — APPROVED** [review]

**Pull request merged**